torch_xla/debug/metrics_test_wrapper.py

```diff
@@ -0,0 +1,99 @@
+"""Wrap PyTorch XLA training jobs with extra logic around XLA metrics.
+
+It can be used to save a historical record of metrics and/or check metrics
+against a golden set of metrics to search for performance regressions.
+
+Example usage:
+python metrics_test_wrapper.py --metrics_output_filename="metrics.txt" \
+    -- python test/test_train_mnist.py --num_epochs=1
+"""
+import argparse
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+
+import torch_xla.utils.gcsfs as gcsfs
+
+
+_CLOUD_STORAGE_PREFIX = 'gs://'
+_XLA_METRICS_FILE = 'XLA_METRICS_FILE'
+
+
+def _run_subprocess(cmd, distributed=False):
+  # Use _XLA_METRICS_FILE to pass the metrics report from the subprocess
+  # to this process. Note that distributed training jobs write 1 file per
+  # core.
+  tmp_dir = tempfile.mkdtemp()
+  _, tmp_path = tempfile.mkstemp(dir=tmp_dir)
+  file_to_read = '{}.0'.format(tmp_path) if distributed else tmp_path
```

Comment:
I'd do a glob(), sort, and pick first.
In case of single core there will be no .0

Reply:
Good idea - switched to that and removed the --distributed flag